session/manager.go

```diff
@@ -47,19 +51,25 @@ type PromiseProcessor interface {
 	Stop() error
 }
 
+// Manager defines methods for session management
+type Manager interface {
```

Comment:
This interface is needed here - caller should define it, not implementer

Reply:
Removed interface. Created Creator interface at caller.